pkg/resources/secondary_database_acceptance_test.go

```diff
@@ -0,0 +1,333 @@
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "external_volume", newExternalVolumeId.Name()),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "catalog", newCatalogId.Name()),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "default_ddl_collation", "en_GB"),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "log_level", string(sdk.LogLevelDebug)),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "trace_level", string(sdk.TraceLevelAlways)),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "comment", newComment),
+				),
+			},
+			{
+				ConfigVariables: configVariables(name, externalPrimaryId, false, sdk.Int(2), sdk.Int(5), externalVolumeId.Name(), catalogId.Name(), "en_US", string(sdk.LogLevelInfo), string(sdk.TraceLevelOnEvent), comment),
+				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_SecondaryDatabase/complete-optionals-set"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "name", name),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "is_transient", "false"),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "as_replica_of", externalPrimaryId.FullyQualifiedName()),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "data_retention_time_in_days.0.value", "2"),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "max_data_extension_time_in_days.0.value", "5"),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "external_volume", externalVolumeId.Name()),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "catalog", catalogId.Name()),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "default_ddl_collation", "en_US"),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "log_level", string(sdk.LogLevelInfo)),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "trace_level", string(sdk.TraceLevelOnEvent)),
+					resource.TestCheckResourceAttr("snowflake_secondary_database.test", "comment", comment),
+				),
+			},
+			// Import all values
+			{
+				ConfigVariables:   configVariables(name, externalPrimaryId, false, sdk.Int(2), sdk.Int(5), externalVolumeId.Name(), catalogId.Name(), "en_US", string(sdk.LogLevelInfo), string(sdk.TraceLevelOnEvent), comment),
+				ConfigDirectory:   acc.ConfigurationDirectory("TestAcc_SecondaryDatabase/complete-optionals-set"),
+				ResourceName:      "snowflake_secondary_database.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAcc_CreateSecondaryDatabase_DataRetentionTimeInDays(t *testing.T) {
```

Comment:
My thinking was that it's the same feature, but in a different place, so I just left it in one place. Especially, because here I'm mostly testing the custom diff functionality.